zerver/lib/actions.py

```diff
@@ -3471,6 +3471,36 @@ def send_peer_subscriber_events(
             send_event(realm, event, peer_user_ids)
 
 
+def send_notification_user_unsubscribe(
+    realm: Realm,
+    stream_dict: Dict[int, Stream],
+    altered_user_dict: Dict[int, Set[int]],
+    private_peer_dict: Dict[int, Set[int]],
+) -> None:
+    private_stream_ids = [
+        stream_id for stream_id in altered_user_dict if stream_dict[stream_id].invite_only
+    ]
+    for stream_id in private_stream_ids:
+        altered_user_ids = altered_user_dict[stream_id]
+        peer_user_ids = private_peer_dict[stream_id] - altered_user_ids
+
+        if peer_user_ids and altered_user_ids:
+            for peer in peer_user_ids:
+                peer_profile = get_user_profile_by_id(peer)
```

Comment:
We try to avoid database queries in a loop. Can you try to find a better way to query this?

Reply:
I have made these changes 
```python
peer_user_profiles_with_enabled_notifications = UserProfile.objects.filter(id__in= peer_user_ids, enable_notification_on_unsubscribe_stream= True)
            altered_user_profiles = UserProfile.objects.filter(id__in= altered_user_ids)
```